waspc/src/Wasp/Project/Vite.hs

```diff
@@ -0,0 +1,37 @@
+module Wasp.Project.Vite where
+
+import Data.List (find)
+import Data.Maybe (fromJust)
+import StrongPath (File', Path', Rel, reldir, relfile, (</>))
+import qualified StrongPath.Operations as SP
+import Wasp.AppSpec.ExternalCode (SourceExternalCodeDir)
+import qualified Wasp.AppSpec.ExternalCode as ExternalCode
+import qualified Wasp.Generator.ExternalCodeGenerator.Common as ECC
+import qualified Wasp.Generator.WebAppGenerator.Common as C
+import qualified Wasp.Generator.WebAppGenerator.ExternalCodeGenerator as EC
+import Wasp.JsImport (JsImport, JsImportName (JsImportModule), makeJsImport)
+
+findCustomViteConfigPath :: [ExternalCode.File] -> Maybe (Path' (Rel SourceExternalCodeDir) File')
+findCustomViteConfigPath externalClientCodeFiles = ExternalCode._pathInExtCodeDir <$> maybeCustomViteConfigPath
+  where
+    maybeCustomViteConfigPath = find isCustomViteConfig externalClientCodeFiles
+
+    isCustomViteConfig :: ExternalCode.File -> Bool
+    isCustomViteConfig
+      ExternalCode.File
+        { _pathInExtCodeDir = path
+        } = path == pathToViteTsConfig || path == pathToViteJsConfig
+
+    pathToViteTsConfig :: Path' (Rel SourceExternalCodeDir) File'
+    pathToViteTsConfig = [relfile|vite.config.ts|]
+
+    pathToViteJsConfig :: Path' (Rel SourceExternalCodeDir) File'
+    pathToViteJsConfig = [relfile|vite.config.js|]
+
+makeCustomViteConfigJsImport :: Path' (Rel SourceExternalCodeDir) File' -> JsImport
```

Comment:
So this should probably go into Generator.
Some indications for this:
1. All the Generator imports in this module are used in this function, so it is heavily connected to Generator.
2. Creating that JS import -> that doesn't sound like general Project level thing, it sounds like something Generator needs.

There is also a bit of suspicious code in it -> I am not sure when that import is generated, where can it be used, in which JS file? Is it absolute, relative to something? If it was in the Generator, in `where` block of place where it is used, that would give it that context. This way as it is now, it feels pulled out of context. That `reldir|src|` is also tiny bit suspicious, feels like that shouldn't be hardcoded here but instead coming from some variable.
I think once you move this to Generator it will all sit in its place.